types/generate.js

```diff
@@ -0,0 +1,114 @@
+// TypeScript definitions are generated here.
+// AVA allows chaining of function names, like `test.after.cb.always`.
+// The order of these names is not important.
+// Writing these definitions by hand is hard. Because of chaining,
+// the number of combinations grows fast (2^n). To reduce this number,
+// illegal combinations are filtered out in `verify`.
+// The order of the options is not important. We could generate full
+// definitions for each possible order, but that would give a very big
+// output. Instead, we write an alias for different orders. For instance,
+// `after.cb` is fully written, and `cb.after` is emitted as an alias
+// using `typeof after.cb`.
+
+var path = require('path');
+var fs = require('fs');
+
+var base = fs.readFileSync(path.join(__dirname, 'base.d.ts')).toString();
+
+// All suported function names
+var allParts = ['serial', 'before', 'after', 'skip', 'todo', 'failing', 'only', 'beforeEach', 'afterEach', 'cb', 'always'];
```

Comment:
Sounds like a good idea, it can be really annoying that the type definitions are not up to date.